src/input/type_record.cc

```diff
@@ -663,6 +721,38 @@ bool AbstractRecord::have_default_descendant() const {
 
 
 
+TypeBase::MakeInstanceReturnType AbstractRecord::make_instance(std::vector<ParameterPair> vec) const {
+	AbstractRecord abstract = this->deep_copy();
+
+	// Set close flag - only for add descendants
+	abstract.child_data_->closed_ = true;
+	// make instances of all descendant records and add them into instance of abstract
+	for (ChildDataIter child_it = begin_child_data(); child_it != end_child_data(); ++child_it) {
+		MakeInstanceReturnType inst = (*child_it).make_instance(vec);
+		abstract.add_child( static_cast<Record &>( *(inst.first) ) );
+		abstract.add_to_parameter_map(inst.second);
+	}
+	// Unset close flag - necessary for set parameters
+	abstract.child_data_->closed_ = false;
+
+	// Set parameters and generic type as attributes
+	abstract.add_attribute("parameters", abstract.print_parameter_map_to_json());
```

Comment:
This code seems to repeat in Array, Record and here, could we make appropriate method in TypeBase.